lib/cogctl/actions/users/info.ex

```diff
@@ -1,41 +1,50 @@
 defmodule Cogctl.Actions.Users.Info do
   use Cogctl.Action, "users info"
-  alias Cogctl.Table
+  import Cogctl.Actions.Users.Util, only: [render: 4]
 
   def option_spec do
     [{:user, :undefined, :undefined, {:string, :undefined}, 'User username (required)'}]
   end
 
+  def run(options, _args, _config, %{token: nil}=endpoint),
+    do: with_authentication(endpoint, &run(options, nil, nil, &1))
   def run(options, _args, _config, endpoint) do
-    with_authentication(endpoint,
-                        &do_info(&1, :proplists.get_value(:user, options)))
-  end
-
-  defp do_info(_endpoint, :undefined) do
-    display_arguments_error
+    case :proplists.get_value(:user, options) do
+      :undefined ->
+        display_arguments_error
+      username ->
+        do_info(endpoint, username)
+    end
   end
 
-  defp do_info(endpoint, user_username) do
-    case CogApi.HTTP.Internal.user_show(endpoint, user_username) do
-      {:ok, resp} ->
-        user = resp["user"]
+  defp do_info(endpoint, username) do
+    case CogApi.HTTP.Client.user_show(endpoint, %{username: username}) do
+      {:ok, user} ->
 
-        user_attr = for {title, attr} <- [{"ID", "id"}, {"Username", "username"}, {"First Name", "first_name"}, {"Last Name", "last_name"}, {"Email", "email_address"}] do
-          [title, user[attr]]
+        user_attr = for {title, attr} <- [{"ID", :id}, {"Username", :username}, {"First Name", :first_name}, {"Last Name", :last_name}, {"Email", :email_address}] do
+          [title, Map.fetch!(user, attr)]
         end
 
-        groups = Enum.map(user["groups"], fn(membership) ->
-                   [membership["name"], membership["id"]]
+        groups = Enum.map(user.groups, fn(membership) ->
+                      [membership.name, membership.id]
                  end)
 
-        display_output("""
-                       #{Table.format(user_attr, false)}
+        role_attr = Enum.reduce(user.groups, [], fn(membership, roles) ->
+                      roles ++ Enum.map(membership.roles, fn(role) ->
+                          permissions = role.permissions |> build_permissions |> Enum.join(",")
+                          [role.name, role.id, permissions]
+                      end)
+                end)
```

Comment:
Probably a good idea to break this up, since there's a lot going on.

It doesn't look like you're separating roles out by the groups they're attached to, so this will repeat entries if you're, say, in two groups that each are granted the same role.

Try getting the unique roles first, then doing an `Enum.map` (because this is fundamentally a `map` operation, not a `reduce`) to generate the table rows.